src/gcp/cloudfunctions.ts

```diff
@@ -260,6 +251,74 @@ export async function setIamPolicy(options: IamOptions): Promise<void> {
   }
 }
 
+/**
+ * Formats the service account to be used with IAM API calls
+ * @param serviceAccount the custom service account created by the user
+ * @param projectId the ID of the current project
+ * @returns a correctly formatted service account string
+ */
+function formatServiceAccount(serviceAccount: string, projectId: string): string {
+  if (serviceAccount.length === 0) {
+    throw new Error("Service account cannot be an empty string");
+  }
+  let emailId = "";
+  if (serviceAccount.charAt(serviceAccount.length - 1) === "@") {
+    emailId = `${serviceAccount}${projectId}.iam.gserviceaccount.com`;
+  } else {
+    emailId = `${serviceAccount}@${projectId}.iam.gserviceaccount.com`;
+  }
+
+  return `serviceAccount:${emailId}`;
+}
+
+/**
+ * Generates the correct IAM Policy for changing the invoker for HTTP triggered functions.
+ * @param projectId the ID of the current project
+ * @param invoker the member(s) that will be granted the invoker role
+ * @returns correctly formatted IAM policy for use in IAM API calls
+ */
+export function generateIamPolicy(
+  projectId: string,
+  invoker: backend.Invoker | backend.Invoker[] = "public"
+): iam.Policy {
+  const invokerRole = "roles/cloudfunctions.invoker";
+  const policy: iam.Policy = {
+    version: 3,
+    bindings: [],
+    etag: "",
+  };
+
+  if (typeof invoker === "string") {
+    if (invoker === "public") {
+      policy.bindings.push({
+        role: invokerRole,
+        members: ["allUsers"],
+      });
+    } else if (invoker === "private") {
+      policy.bindings.push({
+        role: invokerRole,
+        members: [],
+      });
+    } else {
+      policy.bindings.push({
+        role: invokerRole,
+        members: [formatServiceAccount(invoker, projectId)],
+      });
+    }
+  } else {
+    const serviceAccountMembers: string[] = [];
+    invoker.forEach((element) =>
```

Comment:
Prefer for... of... : go/tsstyle#iterating-containers

```
Do not use Array.prototype.forEach, Set.prototype.forEach, and Map.prototype.forEach. 
They make code harder to debug and defeat some useful compiler checks (e.g. reachability).

````

Reply:
If you're feeling this functional style tho, maybe use map instead?
```
else {
  policy.bindings.push({
    role: invokerRole,
    members: invoker.map(e => formatServiceAccount(e, projectId)),
  })
}
```

Reply:
Switched to the for ... of ... syntax